From 084ff8e7fdda7123ca029332410c7d1b9706173a Mon Sep 17 00:00:00 2001
From: Bjoern Bachmann <bjoern.bachmann@emerson.com>
Date: Tue, 21 Apr 2026 13:13:20 +0200
Subject: [PATCH 076/110] arm64: usrp_defconfig: Fixing the embedded tests.

Add debug and tracing options: KGDB, ftrace syscall/scheduler/hwlat
tracers, latency top, stack tracer, and various debug configs.

Enable additional peripheral drivers:
- USB serial adapters (CH341, CP210X, FTDI_SIO, PL2303)
- Xilinx AXI EMAC and DMA
- RPMSG character devices
- EXTCON TUSB320 USB-C
- GPIO sysfs interface
- KEXEC_FILE and NAMESPACES support
---
 arch/arm64/configs/usrp_defconfig | 37 ++++++++++++++++++++++++++++++-
 1 file changed, 36 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/configs/usrp_defconfig b/arch/arm64/configs/usrp_defconfig
index 1014de78e65e5..6278b0cdd93a4 100644
--- a/arch/arm64/configs/usrp_defconfig
+++ b/arch/arm64/configs/usrp_defconfig
@@ -29,11 +29,14 @@ CONFIG_CGROUP_DEVICE=y
 CONFIG_CGROUP_CPUACCT=y
 CONFIG_CGROUP_PERF=y
 CONFIG_CGROUP_BPF=y
+CONFIG_NAMESPACES=y
 CONFIG_USER_NS=y
 CONFIG_CHECKPOINT_RESTORE=y
 CONFIG_SCHED_AUTOGROUP=y
 CONFIG_BLK_DEV_INITRD=y
+CONFIG_EXPERT=y
 CONFIG_PROFILING=y
+CONFIG_KEXEC_FILE=y
 CONFIG_ARCH_ZYNQMP=y
 CONFIG_SCHED_MC=y
 CONFIG_SCHED_SMT=y
@@ -141,9 +144,11 @@ CONFIG_RXKAD=y
 CONFIG_AF_KCM=y
 CONFIG_DEVTMPFS=y
 CONFIG_DEVTMPFS_MOUNT=y
+CONFIG_DEBUG_DEVRES=y
 CONFIG_CONNECTOR=y
 CONFIG_ARM_SCMI_PROTOCOL=y
 CONFIG_ARM_SCPI_PROTOCOL=y
+CONFIG_ZYNQMP_FIRMWARE_DEBUG=y
 CONFIG_MTD=y
 CONFIG_MTD_BLOCK=y
 CONFIG_MTD_SPI_NOR=y
@@ -189,6 +194,8 @@ CONFIG_NI_XGE_MANAGEMENT_ENET=y
 # CONFIG_NET_VENDOR_SYNOPSYS is not set
 # CONFIG_NET_VENDOR_VIA is not set
 # CONFIG_NET_VENDOR_WIZNET is not set
+CONFIG_XILINX_AXI_EMAC=y
+CONFIG_LED_TRIGGER_PHY=y
 CONFIG_MICREL_PHY=y
 # CONFIG_WLAN is not set
 CONFIG_INPUT_EVDEV=y
@@ -228,6 +235,7 @@ CONFIG_SPI_ZYNQMP_GQSPI=y
 CONFIG_SPI_SPIDEV=y
 CONFIG_PINCTRL=y
 CONFIG_GPIOLIB=y
+CONFIG_GPIO_SYSFS=y
 CONFIG_GPIO_ZYNQ=y
 CONFIG_GPIO_PCA953X=y
 CONFIG_GPIO_CROS_EC=y
@@ -244,6 +252,7 @@ CONFIG_REGULATOR=y
 CONFIG_REGULATOR_FIXED_VOLTAGE=y
 # CONFIG_HID is not set
 # CONFIG_USB_HID is not set
+CONFIG_USB_LED_TRIG=y
 CONFIG_USB_ULPI_BUS=y
 CONFIG_USB=y
 CONFIG_USB_ANNOUNCE_NEW_DEVICES=y
@@ -255,11 +264,14 @@ CONFIG_USB_OHCI_HCD=y
 CONFIG_USB_STORAGE=y
 CONFIG_USB_UAS=y
 CONFIG_USB_DWC3=y
-CONFIG_USB_DWC3_XILINX=y
 CONFIG_USB_DWC3_ULPI=y
 CONFIG_USB_SERIAL=y
 CONFIG_USB_SERIAL_CONSOLE=y
 CONFIG_USB_SERIAL_GENERIC=y
+CONFIG_USB_SERIAL_CH341=y
+CONFIG_USB_SERIAL_CP210X=y
+CONFIG_USB_SERIAL_FTDI_SIO=y
+CONFIG_USB_SERIAL_PL2303=y
 CONFIG_USB_ULPI=y
 CONFIG_USB_GADGET=y
 CONFIG_USB_CONFIGFS=y
@@ -277,6 +289,7 @@ CONFIG_LEDS_GPIO=y
 CONFIG_LEDS_TRIGGER_HEARTBEAT=y
 CONFIG_LEDS_TRIGGER_ACTIVITY=y
 CONFIG_LEDS_TRIGGER_GPIO=y
+CONFIG_LEDS_TRIGGER_TRANSIENT=y
 CONFIG_LEDS_TRIGGER_PANIC=y
 CONFIG_LEDS_TRIGGER_NETDEV=y
 CONFIG_EDAC=y
@@ -285,6 +298,7 @@ CONFIG_RTC_CLASS=y
 CONFIG_RTC_DRV_DS1374=y
 CONFIG_RTC_DRV_CROS_EC=y
 CONFIG_DMADEVICES=y
+CONFIG_XILINX_DMA=y
 CONFIG_ASYNC_TX_DMA=y
 CONFIG_SYNC_FILE=y
 CONFIG_UDMABUF=y
@@ -303,8 +317,10 @@ CONFIG_CROS_EC_I2C=y
 CONFIG_HWSPINLOCK=y
 CONFIG_REMOTEPROC=y
 CONFIG_XLNX_R5_REMOTEPROC=y
+CONFIG_RPMSG_CHAR=y
 CONFIG_PM_DEVFREQ=y
 CONFIG_EXTCON=y
+CONFIG_EXTCON_USBC_TUSB320=y
 CONFIG_IIO=y
 CONFIG_IIO_BUFFER=y
 CONFIG_IIO_SW_TRIGGER=y
@@ -338,6 +354,7 @@ CONFIG_FUSE_FS=y
 CONFIG_MSDOS_FS=y
 CONFIG_VFAT_FS=y
 CONFIG_FAT_DEFAULT_IOCHARSET="ascii"
+CONFIG_PROC_VMCORE_DEVICE_DUMP=y
 CONFIG_TMPFS=y
 CONFIG_TMPFS_POSIX_ACL=y
 CONFIG_JFFS2_FS=y
@@ -381,12 +398,30 @@ CONFIG_DMA_CMA=y
 CONFIG_CMA_SIZE_MBYTES=64
 CONFIG_IRQ_POLL=y
 CONFIG_PRINTK_TIME=y
+CONFIG_PRINTK_CALLER=y
 CONFIG_CONSOLE_LOGLEVEL_QUIET=3
 CONFIG_DYNAMIC_DEBUG=y
 CONFIG_STRIP_ASM_SYMS=y
+CONFIG_MAGIC_SYSRQ_DEFAULT_ENABLE=0x0
+CONFIG_KGDB=y
+CONFIG_KGDB_TESTS=y
 CONFIG_DEBUG_RODATA_TEST=y
 CONFIG_DEBUG_WX=y
+CONFIG_DEBUG_VM=y
+# CONFIG_DEBUG_VM_PGTABLE is not set
+CONFIG_DEBUG_MEMORY_INIT=y
+CONFIG_DEBUG_SHIRQ=y
+CONFIG_SOFTLOCKUP_DETECTOR=y
+# CONFIG_DETECT_HUNG_TASK is not set
 CONFIG_RCU_CPU_STALL_TIMEOUT=60
+# CONFIG_RCU_TRACE is not set
+CONFIG_LATENCYTOP=y
+CONFIG_STACK_TRACER=y
+CONFIG_SCHED_TRACER=y
+CONFIG_HWLAT_TRACER=y
+CONFIG_FTRACE_SYSCALLS=y
+CONFIG_BLK_DEV_IO_TRACE=y
+CONFIG_TRACE_EVAL_MAP_FILE=y
 # CONFIG_STRICT_DEVMEM is not set
 CONFIG_ATOMIC64_SELFTEST=y
 CONFIG_TEST_KSTRTOX=y
-- 
2.43.0

